h2o-algos/src/main/java/hex/glm/GLMModel.java

```diff
@@ -109,6 +111,45 @@ public void addScoringInfo(GLMParameters parms, int nclasses, long currTime, int
     }
     scoringInfo = ScoringInfo.prependScoringInfo(currInfo, scoringInfo);
   }
+
+  public ScoringInfo[] getUnrestrictedModelScoringInfo() { return _unrestrictedModelScoringInfo;}
+
+  public ScoreKeeper[] unrestritedModelScoreKeepers() {
+    int size = _unrestrictedModelScoringInfo ==null? 0: _unrestrictedModelScoringInfo.length;
+    ScoreKeeper[] sk = new ScoreKeeper[size];
+    for (int i=0;i<size;++i) {
+      if (_unrestrictedModelScoringInfo[i].cross_validation) // preference is to use xval first, then valid and last train.
+        sk[i] = _unrestrictedModelScoringInfo[i].scored_xval;
+      else if (_unrestrictedModelScoringInfo[i].validation)
+        sk[i] = _unrestrictedModelScoringInfo[i].scored_valid;
+      else
+        sk[i] = _unrestrictedModelScoringInfo[i].scored_train;
+    }
+    return sk;
+  }
+
+  public void addUnrestrictedModelScoringInfo(GLMParameters parms, int nclasses, long currTime, int iter) {
+    if (_unrestrictedModelScoringInfo != null && (((GLMScoringInfo) _unrestrictedModelScoringInfo[_unrestrictedModelScoringInfo.length-1]).iterations() >= iter)) {  // no duplication
+      return;
+    }
```

Comment:
Why is this necessary? This looks like a way to hide a bug, i.e., we should call this just once per iteration.

If anything I'd put there `assert` but not some hack to make sure it works.

If it's called multiple times during the same iteration does it have the same values? Why is it called multiple times?

Reply:
Sure. Here is the issue: #16682.